File: frontend/src/models/Expense.ts
export const ExpenseCategory = {
    Transport: 'Transport',
    Accommodation: 'Accommodation',
    Food: 'Food',
    Tickets: 'Tickets',
    Shopping: 'Shopping',
    Other: 'Other'
} as const;

export type ExpenseCategory = (typeof ExpenseCategory)[keyof typeof ExpenseCategory];

export const ExpenseCategoryLabels: Record<ExpenseCategory, string> = {
    [ExpenseCategory.Transport]: 'Transport',
    [ExpenseCategory.Accommodation]: 'Accommodation',
    [ExpenseCategory.Food]: 'Food',
    [ExpenseCategory.Tickets]: 'Tickets',
    [ExpenseCategory.Shopping]: 'Shopping',
    [ExpenseCategory.Other]: 'Other'
};

export interface Expense {
    id: number;
    travelPlanId: number;
    name: string;
    category: ExpenseCategory;
    amount: number;
    date: string;
    description?: string;
}

export interface ExpenseRequestDto {
    name: string;
    category: ExpenseCategory;
    amount: number;
    date: string;
    description?: string;
}

File: backend/TravelPlanService/Models/Entities/Expense.cs
namespace TravelPlanService.Models.Entities;

public class Expense
{
    public long Id { get; set; }
    public long TravelPlanId { get; set; }
    public string Name { get; set; } = string.Empty;
    public ExpenseCategory Category { get; set; } = ExpenseCategory.Other;
    public decimal Amount { get; set; }
    public DateTime Date { get; set; }
    public string? Description { get; set; }

    public TravelPlan TravelPlan { get; set; } = null!;
}

public enum ExpenseCategory
{
    Transport = 0,
    Accommodation = 1,
    Food = 2,
    Tickets = 3,
    Shopping = 4,
    Other = 5
}
